command/job_history.go

```diff
@@ -118,6 +134,25 @@ func (c *JobHistoryCommand) Run(args []string) int {
 		return 1
 	}
 
+	if (diffTag != "" && !diff) || (diffVersionFlag != "" && !diff) {
+		c.Ui.Error("-diff-tag and -diff-version can only be used with -p")
+		return 1
+	}
+
+	if diffTag != "" && diffVersionFlag != "" {
+		c.Ui.Error("-diff-tag and -diff-version are mutually exclusive")
+		return 1
+	}
+
+	if diffVersionFlag != "" {
+		parsedDiffVersion, err := strconv.ParseUint(diffVersionFlag, 10, 64)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error parsing diff version: %s", err))
```

Comment:
If we show the exact flag they were using that'll be slightly more clear:

```suggestion
			c.Ui.Error(fmt.Sprintf("Error parsing -diff-version: %s", err))
```